### PR TITLE
Added Seqprep to the tools

### PR DESCRIPTION
**Pull request by Alby-Git**
Hello everyone,

I'm submitting a new tool, `seqprep`, to the tools-iuc repository. Below are the details and checklist for this submission:

### Tool details
- **Name:** SeqPrep
- **Description:** SeqPrep is a missing tool in the amplicon pipeline from MGnify, it is designed for merging overlapping paired-end Illumina reads into a single, longer read.
- **Homepage:** [SeqPrep](https://github.com/jstjohn/SeqPrep)
- **License:** [License](https://github.com/jstjohn/SeqPrep/blob/master/LICENSE)

### Checklist for submission
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

Looking forward to your feedback and suggestions.

Best regards,

Albert

(P.S. @RZ9082, @paulzierep - would appreciate your thoughts on this when you have a moment.)

### CODE REVIEW

File: tools/seqprep/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: seqprep
+owner: iuc
+description: Tool for merging paired-end Illumina reads and trimming adapters.
+homepage_url: https://github.com/jstjohn/SeqPrep
+long_description: |
+    SeqPrep is a program designed to merge paired-end Illumina reads that are overlapping into a single longer read and to trim adapters.
+    It is particularly useful for preparing reads for downstream applications like genome assembly and SNP calling.
+    The tool works by identifying and merging reads that are overlapping and trimming adapters based on user-defined sequences or default parameters.
+  remote_repository_url: https://github.com/galaxyproject/tools-iuc/tree/master/tools/seqprep
```
Comment (paulzierep) [inline tools/seqprep/.shed.yml:9]:
```suggestion
remote_repository_url: https://github.com/galaxyproject/tools-iuc/tree/master/tools/seqprep
```

**Comment by paulzierep** [comment]
can you first check the linting error for the shed

**Comment by paulzierep** [comment]
Test 5 has more outputs than expected, and the files are different, can you check that again ? You can see the test results, when you look at the Action Summary (https://github.com/galaxyproject/tools-iuc/actions/runs/7543837079?pr=5719) and then download the Artifacts.

File: tools/seqprep/seqprep.xml [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+            -1 '${output1}'
+            -2 '${output2}'
+
+            #if $merge_reads
+                -s '${merged}'
+            #end if
+
+            ## General Arguments ##
+            #if $general_options.first_read_discarded
+                -3 $general_options.first_read_discarded
```
Comment (paulzierep) [inline tools/seqprep/seqprep.xml:21]:
```suggestion
                -3 '$general_options.first_read_discarded'
```
All parameters need to go in `'`

File: tools/seqprep/seqprep.xml [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+                -m $merging_options.max_mismatch_fraction
+            #end if
+            #if $merging_options.min_match_fraction
+                -n $merging_options.min_match_fraction
+            #end if
+    ]]></command>
+    <inputs>
+        <param label="First Read Input (FASTQ)" name="input1" type="data" format="fastq" />
+        <param label="Second Read Input (FASTQ)" name="input2" type="data" format="fastq" />
+        <param label="Merge Reads" name="merge_reads" type="boolean" truevalue="true" falsevalue="false" checked="true"/>
```
Comment (paulzierep) [inline tools/seqprep/seqprep.xml:103]: help

File: tools/seqprep/seqprep.xml [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+            #end if
+            #if $merging_options.max_mismatch_fraction
+                -m $merging_options.max_mismatch_fraction
+            #end if
+            #if $merging_options.min_match_fraction
+                -n $merging_options.min_match_fraction
+            #end if
+    ]]></command>
+    <inputs>
+        <param label="First Read Input (FASTQ)" name="input1" type="data" format="fastq" />
```
Comment (paulzierep) [inline tools/seqprep/seqprep.xml:101]: does it also work with fastq.gz ?

File: tools/seqprep/seqprep.xml [hunk 5]
```diff
@@ -0,0 +1,327 @@
+<tool id="seqprep" name="Merging paired-end Illumina reads (SeqPrep)" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@">
```
Comment (paulzierep) [inline tools/seqprep/seqprep.xml:1]: can you add a profile 22.05 should be ok

File: tools/seqprep/macros.xml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<macros>
+    <token name="@TOOL_VERSION@">1.3.2</token>
+    <token name="@VERSION_SUFFIX@">8</token>
```
Comment (paulzierep) [inline tools/seqprep/macros.xml:3]: suffix should be 0 I think

**Review by paulzierep — COMMENTED** [review]
minor changes please

File: tools/seqprep/macros.xml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+        </requirements>
+    </xml>
+    <xml name="biotools">
+        <xrefs>
+            <xref type="bio.tools">seqprep</xref>
+        </xrefs>
+    </xml>
+    <xml name="citations">
+        <citations>
+            <citation type="doi">10.1134/S1021443716020175</citation>
```
Comment (RZ9082) [inline tools/seqprep/macros.xml:16]: False citation!

**Comment by paulzierep** [comment]
We just found there is a wrapper for seqprep already: https://github.com/galaxy-works/shed-tools/blob/main/tools/seqprep/seqPrep.xml ... bummer; can you check if this is pretty much identical... i.e. compare some of the tests, then we can ask to deprecate the other repo and give credits to them as well

**Comment by Alby-Git** [comment]
> We just found there is a wrapper for seqprep already: https://github.com/galaxy-works/shed-tools/blob/main/tools/seqprep/seqPrep.xml ... bummer; can you check if this is pretty much identical... i.e. compare some of the tests, then we can ask to deprecate the other repo and give credits to them as well

Ah, bummer. But no problem, it was an amazing opportunity for me to get started with Planemo and understand the tool wrapping process. I will look into it and make the necessary changes. How would I give credit to them? Is there a way to link to them?

**Comment by bgruening** [comment]
@Alby-Git there is a high chance your tool is actually more up to date. Thanks for your work!

**Comment by paulzierep** [comment]
look good from my side @bgruening can we merge this ?

File: tools/seqprep/seqprep.xml [hunk 8]
```diff
@@ -0,0 +1,348 @@
...
+        #if $merging_options.max_mismatch_fraction
+            -m '${merging_options.max_mismatch_fraction}'
+        #end if
+        #if $merging_options.min_match_fraction
+            -n '${merging_options.min_match_fraction}'
+        #end if
+    ]]></command>
+    <inputs>
+        <param label="First Read Input (FASTQ)" name="input1" type="data" format="fastq" help="Select the FASTQ file containing the first set of paired-end reads." />
+        <param label="Second Read Input (FASTQ)" name="input2" type="data" format="fastq" help="Select the FASTQ file containing the second set of paired-end reads." />
```
Comment (bgruening) [inline tools/seqprep/seqprep.xml:102]:
```suggestion
        <param label="First Read Input" name="input1" type="data" format="fastq" help="Select the FASTQ file containing the first set of paired-end reads." />
        <param label="Second Read Input" name="input2" type="data" format="fastq" help="Select the FASTQ file containing the second set of paired-end reads." />
```

File: tools/seqprep/seqprep.xml [hunk 9]
```diff
@@ -0,0 +1,348 @@
...
+
+    **Outputs**
+
+    SeqPrep generates outputs in gzipped FASTQ format.
+
+    See more details on SeqPrep github repository: https://github.com/jstjohn/SeqPrep
+
+    **License**
+
+    SeqPrep is copyrighted to John St. John.
```
Comment (bgruening) [inline tools/seqprep/seqprep.xml:339]: Its MIT licensed. It think this can be removed.

File: tools/seqprep/seqprep.xml [hunk 10]
```diff
@@ -0,0 +1,348 @@
...
+
+    This wrapper is copyrighted by Lionel Guy, and is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
```
Comment (bgruening) [inline tools/seqprep/seqprep.xml:341]: I would remove that and add a https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-creator-person and put you and Lionel ...

File: tools/seqprep/seqprep.xml [hunk 11]
```diff
@@ -0,0 +1,348 @@
...
+
+    This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
```
Comment (bgruening) [inline tools/seqprep/seqprep.xml:343]: This looks more like https://github.com/jstjohn/SeqPrep/blob/master/LICENSE

I would remove this from the help.

**Comment by paulzierep** [comment]
@Alby-Git can you add those changes ? Or are out finish at the moment, then maybe @RZ9082 can take over ?

**Comment by Alby-Git** [comment]
@paulzierep did not get any notifications on that issues sorry. Will fix it today! :')

**Review by bgruening — APPROVED** [review]

**Comment by paulzierep** [comment]
@bgruening can we merge it ?

**Comment by bgruening** [comment]
Thanks @Alby-Git!

**Pull request merged**